Add Imagination's Pistachio SoC and (Ci40) Marduk platform support

abhijit-mahajani-imgtec opened the pull request:
1.add Imagination's pistachio soc support

Also add profile support for Creator Ci40(marduk) platform

2.pistachio: add dts

3.pistachio: add config-4.4

4.pistachio: add kernel-4.4 patches

5.pistachio: add base-files for Ci40

6.uboot-envtools: add pistachio support 

7.Cherry-picked commit from https://github.com/openwrt/openwrt/pull/101

Ci40 H/W documentation available at https://docs.creatordev.io/ci40/guides/hardware/

Contributors:

Abhijit Mahajani
Avinash Tahakik
Francois Berder
Ian Pozella
Manohar Narkhede
Mayank Sirotiya
Pratik Prajapati
Rahul Daga
Sean Kelly
Shraddha Chaudhari


Apart from changing from legacy uImage to fitImage in this Pull Request, it also addresses some of the review comments from #95 
1. All License header changed to GPLV2
2. Added pistachio support in uboot-envtools
3. Added 02_network to read mac address from uboot env variables currently just ethaddr (for now it assumes "ethaddr" is set into uboot env)
4. Removed wifi and lowpan configuration for first basic PR
5. Removed asound.conf from pistachio/base-files. audio support is working, but mic support will be added separately in alsa-lib patches.
6. Cleaned up config-4.4 to remove "builtin as modules" options
7. Removed creator-platform-default config file.

Outstanding review comments:
1. Add wifi support
2. Add bluetooth support
3. Add 6lowpan support
4. Mic support patch into alsa-lib
5. Storing mac addresses into OTP via uboot

Commit:
pistachio: add new target for Imagination's Pistachio SoC

Also add profile support for Creator Ci40(marduk) platform
based upon pistachio SoC

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>
Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>
Signed-off-by: Francois Berder <francois.berder@imgtec.com>

Commit:
pistachio: add dts

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

Commit:
pistachio: add base-files for Ci40

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

Commit:
pistachio: add config-4.4

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>
Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>
Signed-off-by: Francois Berder <francois.berder@imgtec.com>
Signed-off-by: Michal Tusnio <michal.tusnio@imgtec.com>

Commit:
pistachio: add kernel patches for 4.4

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

Ham22 committed:
image.mk: use LINUX_KARCH rather than ARCH for fit

Passing ARCH (e.g. mipsel) to the mkits script causes failures while
using LINUX_KARCH (e.g. mips) works as expected.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

Commit:
uboot-envtools: add pistachio support

Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

Ham22 committed:
pistachio: revert EOF hacks

Currently the upstream build system globally adds EOF markers too
all images to get around issues for systems that have factory
installed images that don't support UBI. However our factory
installed image does support UBI as does our bootloader which
causes problems for booting due to these EOF additions.

Signed-off-by: Ian Pozella <Ian.Pozella@imgtec.com>

francois-berder committed:
kernel: add missing symbol for 4.4

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

wigyori reviewed (commented):
+define KernelPackage/uccp420wlan

The wifi module is still there in modules.mk - can you review or remove it until it's resolved please. Same goes for the 6LoWPAN modules. You can add these back later in a separate PR.

abhijit-mahajani-imgtec commented:
`+define KernelPackage/uccp420wlan

The wifi module is still there in modules.mk - can you review or remove it until it's resolved please. Same goes for the 6LoWPAN modules. You can add these back later in a separate PR.`

Removed uccp420wlan and other 6lowpan , nhc modules from modules.mk

wigyori reviewed (commented):
```
image.mk: use LINUX_KARCH rather than ARCH for fit
```

While this is a straight-forward change, it should be in a separate PR as it affects multiple (all) targets, and has nothing to do with the Pistachio support - please move this, thanks.

wigyori reviewed (commented):
```
add Imagination's pistachio soc support 
```

To me this looks OK at this time ( @lperkov / @hauke please comment), but please update the commit title to something like "pistachio: add new target for Imagination's Pistachio SoC", thanks.

abhijit-mahajani-imgtec commented:
> image.mk: use LINUX_KARCH rather than ARCH for fit
While this is a straight-forward change, it should be in a separate PR as it affects multiple (all) targets, and has nothing to do with the Pistachio support - please move this, thanks.

This is already raised in https://github.com/openwrt/openwrt/pull/101. I guess if that gets merged first then i can rebase all these changes on top of that.
Thanks.

abhijit-mahajani-imgtec commented:
> update the commit title to something like "pistachio: add new target for Imagination's Pistachio SoC"

Done

abhijit-mahajani-imgtec commented:
@lperkov , @hauke any comments please?

Commit:
pistachio: add alias for wifi in dts

to be able to read the mac address and calibration data in u-boot

Signed-off-by: Francois Berder <francois.berder@imgtec.com>
Signed-off-by: Abhijit Mahajani <Abhijit.Mahajani@imgtec.com>

francois-berder committed:
generic/config-4.4: Add missing options

These options were added after adding patch 120 for the pistachio
target which enable generic DMA to avoid using bounce buffers.

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

nbd168 commented on target/linux/pistachio/patches-4.4/0000-2-revert-generic-mtd-ubi-add-EOF-marker-support.patch:
```diff
@@ -0,0 +1,53 @@
+reverted:
```
This is the wrong way to deal with this issue. Even when you revert the patch, the EOF marker is still written to flash as part of the UBI image. If you don't write the EOF marker, the kernel patch is completely harmless.
This really should be fixed in the image building code. 
In LEDE this has been cleaned up properly and EOF markers are only added for devices that actually need them. Please backport those changes in to OpenWrt instead of adding patches like these

nbd168 commented:
The kernel patches seem rather messy to me - containing patches that introduce non-upstream code with bugs and then later patches that fix the bugs or do cleanups.
Since these patches do not seem to be based on upstream commit history, it would be good to have this cleaned up by folding those kinds of patches together.
There are also some kernel patches that touch core subsystem code or MIPS architecture code. Those should be submitted upstream and the patches annotated with a link to patchwork or something like that.

nbd168 commented:
Another thing, the target makefile should probably be added as the last commit, not the first.

nbd168 commented on target/linux/pistachio/Makefile:
```diff
@@ -0,0 +1,26 @@
+BOARD:=pistachio
+BOARDNAME:=IMG MIPS Pistachio
+FEATURES:=targz nand squashfs ubifs audio ramdisk fpu usb usbgadget
+SUBTARGETS=marduk
```
Do you have any plans to add more subtargets soon? If not, you should drop this line.

blogic commented:
i guess with the current state of imgtec this can be closed
